pvnet/models/multimodal/validation/config_static_validation.py

```diff
@@ -0,0 +1,514 @@
+                    f"{message} (Action: May use default value if applicable)."
+                )
+        else:
+            _check_key(
+                cfg,
+                param_name,
+                required=False,
+                expected_type=int,
+                warn_on_type_mismatch=True,
+                context=f"Parameter '{param_name}' associated with '{owner_key}'",
+            )
+
+
+def _validate_nwp_specifics(cfg: dict[str, Any], nwp_section: dict[str, Any]) -> None:
+    """Validate NWP-specific time/interval dictionaries against NWP sources.
+
+    Args:
+        cfg: The main configuration dictionary.
+        nwp_section: The validated 'nwp_encoders_dict' section content.
+
+    Raises:
+        KeyError: If required NWP time/interval parameter dictionaries are missing.
+        TypeError: If NWP time/interval parameters are not dictionaries.
+        ValueError: If keys in time/interval dicts do not match NWP sources.
+    """
+    nwp_sources: list[str] = list(nwp_section.keys())
+    if not nwp_sources:
+        logger.warning("'nwp_encoders_dict' is defined but contains no NWP sources.")
+        return
+
+    context = f"Config includes 'nwp_encoders_dict' with sources: {nwp_sources}"
+
+    # Check time param values are integers
+    _check_key(
+        cfg, "nwp_history_minutes", required=True, expected_type=dict, context=context
+    )
+    _check_key(
+        cfg, "nwp_forecast_minutes", required=True, expected_type=dict, context=context
+    )
+    _check_key(
+        cfg, "nwp_interval_minutes", required=True, expected_type=dict, context=context
+    )
+
+    nwp_hist_times: dict[str, Any] = cfg["nwp_history_minutes"]
+    nwp_forecast_times: dict[str, Any] = cfg["nwp_forecast_minutes"]
+
+    hist_keys: set[str] = set(nwp_hist_times.keys())
+    forecast_keys: set[str] = set(nwp_forecast_times.keys())
+    encoder_keys: set[str] = set(nwp_sources)
+
+    # Verify time params are provided for specifically defined NWP sources
+    if hist_keys != encoder_keys:
+        missing_in_hist = encoder_keys - hist_keys
+        extra_in_hist = hist_keys - encoder_keys
+        raise ValueError(
+            f"Keys in 'nwp_history_minutes' {hist_keys} do not match sources "
+            f"in 'nwp_encoders_dict' {encoder_keys}. "
+            f"Missing: {missing_in_hist}, Extra: {extra_in_hist}"
+        )
+    if forecast_keys != encoder_keys:
+        missing_in_forecast = encoder_keys - forecast_keys
+        extra_in_forecast = forecast_keys - encoder_keys
+        raise ValueError(
+            f"Keys in 'nwp_forecast_minutes' {forecast_keys} do not match sources "
+            f"in 'nwp_encoders_dict' {encoder_keys}. "
+            f"Missing: {missing_in_forecast}, Extra: {extra_in_forecast}"
+        )
+
+    _check_key(
+        cfg, "nwp_interval_minutes", required=True, expected_type=dict, context=context
+    )
+    nwp_intervals: dict[str, Any] = cfg["nwp_interval_minutes"]
```

Comment:
```suggestion
    nwp_intervals: dict[str, int] = cfg["nwp_interval_minutes"]
```

I think this is required to be an int